### ISSUE TITLE
`UndefVarError: TakeWhile not defined` - afterwards REPL results lag behind

### ISSUE THREAD

**Issue by jkrumbiegel** [timholy/Revise.jl#617]
This started happening recently, I'll come back to the REPL after some time and the next statement will throw the following error. Afterwards, each execution in the REPL returns the result of the previous statement which looks like this

```julia
julia> 1 + 1
2 # from before

julia> 1 + 2
2

julia> 1 + 3
3

julia> "hello"
4

julia> "lag"
"hello"
```

This is the error message:

```julia
ERROR: UndefVarError: TakeWhile not defined
Stacktrace:
 [1] top-level scope
   @ ~\.julia\packages\Compat\QxKrF\src\iterators.jl:42
Revise evaluation error at ┌ Warning: C:\Users\Krumbiegel\.julia\packages\IntervalSets\4SvUi\ exists, deleted all methods
└ @ Revise C:\Users\Krumbiegel\.julia\packages\Revise\VxkZO\src\packagedef.jl:650
C:\Users\Krumbiegel\.julia\packages\Compat\QxKrF\src\iterators.jl:42

Stacktrace:
  [1] methods_by_execution!(recurse::Any, methodinfo::Revise.CodeTrackingMethodInfo, docexprspr}}, mod::Module, ex::Expr; mode::Symbol, disablebp::Bool, always_rethrow::Bool, kwargs::Bas}, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
    @ Revise ~\.julia\packages\Revise\VxkZO\src\lowered.jl:231
  [2] #eval_with_signatures#92
    @ ~\.julia\packages\Revise\VxkZO\src\packagedef.jl:463 [inlined]
  [3] instantiate_sigs!(modexsigs::OrderedCollections.OrderedDict{Module, OrderedCollections.atableExpr, Union{Nothing, Vector{Any}}}}; mode::Symbol, kwargs::Base.Iterators.Pairs{Union{}dTuple{(), Tuple{}}})
    @ Revise ~\.julia\packages\Revise\VxkZO\src\packagedef.jl:471
  [4] instantiate_sigs!
    @ ~\.julia\packages\Revise\VxkZO\src\packagedef.jl:468 [inlined]
  [5] maybe_extract_sigs!
    @ ~\.julia\packages\Revise\VxkZO\src\pkgs.jl:145 [inlined]
  [6] (::Revise.var"#44#45"{String})()
    @ Revise ~\.julia\packages\Revise\VxkZO\src\pkgs.jl:518
  [7] with_logstate(f::Function, logstate::Any)
    @ Base.CoreLogging .\logging.jl:491
  [8] with_logger
    @ .\logging.jl:603 [inlined]
  [9] watch_manifest(mfile::String)
    @ Revise ~\.julia\packages\Revise\VxkZO\src\pkgs.jl:497
 [10] (::Revise.TaskThunk)()
    @ Revise ~\.julia\packages\Revise\VxkZO\src\types.jl:272

caused by: UndefVarError: TakeWhile not defined
Stacktrace:
 [1] macro expansion
   @ ~\.julia\packages\JuliaInterpreter\lObtQ\src\interpret.jl:60 [inlined]
 [2] getargs(args::Vector{Any}, frame::JuliaInterpreter.Frame)
   @ JuliaInterpreter ~\.julia\packages\JuliaInterpreter\lObtQ\src\builtins.jl:8
 [3] maybe_evaluate_builtin(frame::JuliaInterpreter.Frame, call_expr::Expr, expand::Bool)
   @ JuliaInterpreter ~\.julia\packages\JuliaInterpreter\lObtQ\src\builtins.jl:116
 [4] evaluate_call_recurse!(recurse::Any, frame::JuliaInterpreter.Frame, call_expr::Expr; ent
   @ JuliaInterpreter ~\.julia\packages\JuliaInterpreter\lObtQ\src\interpret.jl:204
 [5] evaluate_call_recurse!
   @ ~\.julia\packages\JuliaInterpreter\lObtQ\src\interpret.jl:201 [inlined]
 [6] eval_rhs(recurse::Any, frame::JuliaInterpreter.Frame, node::Expr)
   @ JuliaInterpreter ~\.julia\packages\JuliaInterpreter\lObtQ\src\interpret.jl:395
 [7] step_expr!(recurse::Any, frame::JuliaInterpreter.Frame, node::Any, istoplevel::Bool)
   @ JuliaInterpreter ~\.julia\packages\JuliaInterpreter\lObtQ\src\interpret.jl:533
 [8] methods_by_execution!(recurse::Any, methodinfo::Revise.CodeTrackingMethodInfo, docexprs:r}}, frame::JuliaInterpreter.Frame, isrequired::Vector{Bool}; mode::Symbol, skip_include::Boo
   @ Revise ~\.julia\packages\Revise\VxkZO\src\lowered.jl:440
```

```julia
Julia Version 1.6.0-rc1
Commit a58bdd9010 (2021-02-06 15:49 UTC)
Platform Info:
  OS: Windows (x86_64-w64-mingw32)
  CPU: Intel(R) Xeon(R) W-2265 CPU @ 3.50GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-11.0.1 (ORCJIT, cascadelake)
```

**Comment by racinmat** [comment]
Happens to me too, on Julia 1.5.4 and Revise 3.1.1.

**Comment by timholy** [comment]
Most likely, in some other Julia process you updated your packages. Revise tries to handle such cases, but when types appear or disappear in different package versions it sometimes goes awry.

If you do find a reproducible test case I will see what I can do.

**Comment by MarcMush** [comment]
I had a similar issue, with `TakeWhile` not defined and, `Revise` failing to revise a lot of packages.
It happened when displaying an error, and it happened each time, meaning it's probably not due to an other process updating Julia

`Pkg.up()` fixed the issue, so I can't reproduce it anymore though :-/

**Comment by MarcMush** [comment]
update: it happened again, but it was not exactly the same thing

I can't replicate it at all though (sigh)
This time there wasn't an error with `TakeWhile` but a world age problem, Revise was trying to access a function in the future

This time I was using Julia in an other terminal as well, so it might be a change somewhere in a package (but I didn't load any package except OhMyREPL and Revise which load at startup, and all I did was checking the docs of a function in base Julia)

additional complexity that might or might not influence the bug:
- I have different versions of packages in the project where it happened than in my main Julia environment, including one that is `dev` in my main env and `add` in the project
- the project uses Distributed and it's a pain to load an environment on all workers (but I checked, all workers have the same package versions)

**Comment by timholy** [comment]
The REPL lag may be fixed by #649, but I don't know how to test it.
